sdjournal/journal.go

```diff
@@ -84,13 +260,49 @@ func (m *Match) String() string {
 	return m.Field + "=" + m.Value
 }
 
+func (j *Journal) getFunction(name string) (unsafe.Pointer, error) {
+	j.mu.Lock()
+	f, ok := libsystemdFunctions[name]
+	if !ok {
+		var err error
+		f, err = j.lib.GetSymbolPointer(name)
+		if err != nil {
+			return nil, err
+		}
+
+		libsystemdFunctions[name] = f
+	}
+	j.mu.Unlock()
+
+	return f, nil
+}
+
 // NewJournal returns a new Journal instance pointing to the local journal
-func NewJournal() (*Journal, error) {
-	j := &Journal{}
-	r := C.sd_journal_open(&j.cjournal, C.SD_JOURNAL_LOCAL_ONLY)
+func NewJournal() (j *Journal, err error) {
+	h, err := dlopen.GetHandle(libsystemdNames)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err != nil {
+			err2 := h.Close()
+			if err2 != nil {
+				err = fmt.Errorf(`%q and "error closing handle: %v"`, err, err2)
```

Comment:
I don't really like the errwrap solution because we're not using it in go-systemd. I also don't like ignoring the closing error because we'd leak a handler, not the same case as with `file`...